opm/autodiff/BlackoilAquiferModel.hpp

```diff
@@ -34,45 +36,61 @@ namespace Opm {
 
         /// Class for handling the blackoil well model.
         template<typename TypeTag>
-        class BlackoilAquiferModel {
-
-        public:
-
-            // ---------      Types      ---------
-            typedef typename GET_PROP_TYPE(TypeTag, ElementContext)      ElementContext;
-            typedef typename GET_PROP_TYPE(TypeTag, Simulator)           Simulator;
-            typedef typename GET_PROP_TYPE(TypeTag, Scalar)              Scalar;
+        class BlackoilAquiferModel : public Ewoms::EclBaseAquiferModel<TypeTag>
+        {
+            typedef Ewoms::EclBaseAquiferModel<TypeTag> ParentType;
 
-            typedef AquiferCarterTracy<TypeTag> Aquifer_object;
+            typedef typename GET_PROP_TYPE(TypeTag, Simulator) Simulator;
+            typedef typename GET_PROP_TYPE(TypeTag, RateVector) RateVector;
 
+        public:
             explicit BlackoilAquiferModel(Simulator& ebosSimulator);
 
-            // compute the well fluxes and assemble them in to the reservoir equations as source terms
-            // and in the well equations.
-            void assemble( const SimulatorTimerInterface& timer,
-                           const int iterationIdx                );
-
-            // called at the end of a time step
-            void timeStepSucceeded(const SimulatorTimerInterface& timer);
+            void initialSolutionApplied()
+            {
```

Comment:
on it. this is a bit difficult for the aquifer code because that code is inconsistent within the same file and also between files.

Reply:
Yes, this is a problem and everyone gets lost and feels troublesome. 

Let us get a unified coding guideline, then there is no this kind of problems anymore. Then we will have a beautiful coding style cross all the OPM repos, everyone is happy and healthy. New comers will be pleasant to join since it is easy to code with.

Reply:
I don't think that this is our largest problem at the moment, nor that it should be assigned a high priority...

Reply:
also, I'm strongly in favour of the rule that whoever advocated for a given convention needs to be willing to adapt all existing code of all OPM modules to it.

Reply:
> also, I'm strongly in favour of the rule that whoever advocated for a given convention needs to be willing to adapt all existing code of all OPM modules to it.

I think clang-format may be up to the task... It could happen!

Reply:
> I don't think that this is our largest problem at the moment, nor that it should be assigned a high priority..

It is not the most important thing to do, while it comes with you everyday, especially when you want to do some more important things. 

> also, I'm strongly in favour of the rule that whoever advocated for a given convention needs to be willing to adapt all existing code of all OPM modules to it.

Okay. Let us get the guideline first. You do not need to get your hands dirty to do the labor.  How does this sound like? As long as we get the guideline, things will become better. Now, things are only getting worse.